boskos/boskos.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	"k8s.io/test-infra/boskos/server"
+)
+
+var configPath = flag.String("config", "resources.json", "Path to init resource file")
+var storage = flag.String("storage", "", "Path to presistent volume to load the state")
+
+func main() {
+	flag.Parse()
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+
+	r, err := server.NewRanch(*configPath, *storage)
+	if err != nil {
+		logrus.WithError(err).Fatal("Fail to create my ranch! Config: %v, storage : %v", *configPath, *storage)
+	}
+
+	http.Handle("/", handleDefault(r))
+	http.Handle("/acquire", handleAcquire(r))
+	http.Handle("/release", handleRelease(r))
+	http.Handle("/reset", handleReset(r))
+	http.Handle("/update", handleUpdate(r))
+	http.Handle("/metric", handleMetric(r))
+
+	go func() {
+		logTick := time.NewTicker(time.Minute).C
+		saveTick := time.NewTicker(time.Minute).C
+		configTick := time.NewTicker(time.Minute * 10).C
+		for {
+			select {
+			case <-logTick:
+				r.LogStatus()
+			case <-saveTick:
+				r.SaveState()
+			case <-configTick:
+				r.SyncConfig(*configPath)
+			}
+		}
+	}()
+
+	logrus.Info("Start Service")
+	logrus.WithError(http.ListenAndServe(":8080", nil)).Fatal("ListenAndServe returned.")
+}
+
+//  handleDefault: Handler for /, always pass with 200
+func handleDefault(r *server.Ranch) http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		logrus.WithField("handler", "handleDefault").Infof("From %v", req.RemoteAddr)
+	}
+}
+
+//  handleAcquire: Handler for /acquire
```

Comment:
Might also mention that this needs to be POST (and expected verb for the others)